src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -364,7 +364,7 @@ internal int GetSubmissionSlotIndex()
             if (_lazySubmissionSlotIndex == SubmissionSlotIndexToBeAllocated)
             {
                 // TODO (tomat): remove recursion
-                int lastSlotIndex = ScriptCompilationInfo.PreviousScriptCompilation?.GetSubmissionSlotIndex() ?? 0;
+                int lastSlotIndex = ScriptCompilationInfo!.PreviousScriptCompilation?.GetSubmissionSlotIndex() ?? 0;
```

Comment:
I'm not following why this is safe?

Reply:
`ScriptCompilationInfo` is null when we're dealing with a compilation that's not for a script. But here we are (`_lazySubmissionSlotIndex` has a value)